packages/core/versioning/core-version-http-server/src/version_http_toolkit.ts

```diff
@@ -154,12 +111,26 @@ export interface VersionedRouter<Ctx extends RqCtx = RqCtx> {
   options: VersionedRouteRegistrar<'options', Ctx>;
 }
 
+/** @experimental */
+type InValidation<P, Q, B> = RouteValidatorFullConfig<P, Q, B>;
+
+/**
+ * Versioned route validation
+ * @experimental
+ */
+interface FullValidationConfig<P, Q, B, R> {
+  /** Validation to run against route inputs: params, query and body */
+  in?: InValidation<P, Q, B>;
+  /** Validation to run against route output */
+  out?: Type<R>;
+}
```

Comment:
Adding something like this gives us a few new capabilities:

1. TS checking enforcing shape of return type
2. Runtime checks against the shape of return type
3. A clearer path to true HTTP API introspection since we have the types for the return value of the endpoint...